rdm/0006-profiles-and-flavours.md

```diff
@@ -0,0 +1,104 @@
+stakeholders while still being able to develop and use domain authorities **within**
+the InvenioRDM context. I have a dedicated mechanism and space to insert
+customizations shared by instances for my domain this way.
+
+
+## Terminology
+
+**flavour**: A *flavour* represents the main use case of the application
+(e.g., ILS or RDM for now) and it translates to a different packaged application.
+The flavour packaged applications are completely different, which calls
+for a different project repository to define them.
+
+**profiles**: *Profiles* are different domains of interest (e.g., Generic or BioMed)
+for the InvenioRDM *flavour* and translate to different configurations,
+data models, and customizations. Those are not major enough to warrant being
+part of different repositories however. The profiles are defined in the same
+repository side-by-side.
+
+
+## Detailed design
+
+```console
+invenio-cli init --flavour RDM --profile biomed
+```
+
+(or something close to this syntax) generates the pre-configured RDM application
+with the specific resource types, data model, UI and other configuration relevant
+to Bio-Medical institutions. Passing `--profile generic` generates one with
+default configurations.
+
+The [invenio-app-rdm](https://github.com/inveniosoftware/invenio-app-rdm) and/or
+the [invenio-scripts](https://github.com/inveniosoftware/invenio-scripts)
+repository should contain profile definitions and means of generating them
+respectively.
+
+The [invenio-app-ils](https://github.com/inveniosoftware/invenio-app-ils) contains
+the ILS application flavour.
+
+
+## How we teach this
+
+This might be challenging as the terms `profile` and `flavour` have close meaning
+in every day language. Consistently adhering to the correct word for the correct
+technical meaning will be important to anchor that relationship, especially in
+person-to-person, chat and documentation settings.
+
+The terminology should be brought up in the [invenio-cli](https://github.com/inveniosoftware/invenio-scripts) app docs.
+The usage should be demonstrated in the general RDM documentation. `flavour`
+should also be described in [invenio-app-rdm](https://github.com/inveniosoftware/invenio-app-rdm).
+
+
+## Drawbacks
+
+As this is the start of the project, what qualifies as profile-worthy
+and what qualifies as flavour-worthy differences will need to be felt out from
+experience. Luckily, at first, only the RDM profile will be supported.
+
+The implied differences between flavours may be overstated and provide decision
+uncertainty as more flavours are added. Lindsay works in a biology research
+institution. She may be unsure about using a "BioMed" profile versus a generic
+one. Should she create her own profile? It should be communicated that
+further customization is possible to match her setting; the decision is not
+irrevocable. Reconfiguration can be done. It should also be made clear, she
```

Comment:
Are we supporting automatic (e.g. command) reconfigure? This should be added to requirements changed in the invenio-cli if so.

Reply:
I don't think automatic reconfiguration should be supported at this point. Manual reconfiguration should be allowed, but that's just... configuration ;).